miku-server/src/main/scala/org/http4s/miku/MikuBuilder.scala

```diff
@@ -0,0 +1,492 @@
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.netty.handler.codec.http.{
+  HttpContentDecompressor,
+  HttpRequestDecoder,
+  HttpResponseEncoder
+}
+import io.netty.handler.ssl.SslHandler
+import io.netty.handler.timeout.IdleStateHandler
+import org.http4s.HttpService
+import org.http4s.server._
+import org.http4s.util.bug
+import org.log4s.getLogger
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
+
+sealed trait NettyTransport
+case object Jdk extends NettyTransport
+case object Native extends NettyTransport
+
+/** Netty server builder.
+  *
+  *
+  * @param httpService
+  * @param socketAddress
+  * @param idleTimeout
+  * @param maxInitialLineLength The maximum length of the initial line. This effectively restricts the maximum length of a URL that the server will
+  *                             accept, the initial line consists of the method (3-7 characters), the URL, and the HTTP version (8 characters),
+  *                             including typical whitespace, the maximum URL length will be this number - 18.
+  * @param maxHeaderSize The maximum length of the HTTP headers. The most common effect of this is a restriction in cookie length, including
+  *                      number of cookies and size of cookie values.
+  * @param maxChunkSize The maximum length of body bytes that Netty will read into memory at a time.
+  *                     This is used in many ways.  Note that this setting has no relation to HTTP chunked transfer encoding - Netty will
+  *                     read "chunks", that is, byte buffers worth of content at a time and pass it to Play, regardless of whether the body
+  *                     is using HTTP chunked transfer encoding.  A single HTTP chunk could span multiple Netty chunks if it exceeds this.
+                        A body that is not HTTP chunked will span multiple Netty chunks if it exceeds this or if no content length is
+                        specified. This only controls the maximum length of the Netty chunk byte buffers.
+  * @param sslBits The SSL information.
+  * @param serviceErrorHandler
+  * @param transport
+  * @param ec
+  * @param banner
+  * @param F
+  * @tparam F
+  */
+class MikuBuilder[F[_]](
+    httpService: HttpService[F],
+    socketAddress: InetSocketAddress,
+    idleTimeout: Duration,
+    maxInitialLineLength: Int,
+    maxHeaderSize: Int,
+    maxChunkSize: Int,
+    sslBits: Option[MikuSSLConfig],
+    serviceErrorHandler: ServiceErrorHandler[F],
+    transport: NettyTransport,
+    ec: ExecutionContext,
+    enableWebsockets: Boolean,
+    banner: immutable.Seq[String],
+    nettyChannelOptions: MikuBuilder.NettyChannelOptions
+)(implicit F: Effect[F])
+    extends ServerBuilder[F]
+    with IdleTimeoutSupport[F]
+    with SSLKeyStoreSupport[F]
```

Comment:
I'm still thinking this needs to go away, but I guess it's consistent with 0.18.

Reply:
I can look into it for master branch. I can create + self assign a ticket